pkg/kubectl/resource/builder_test.go

```diff
@@ -608,6 +609,26 @@ func TestMultipleResourceByTheSameName(t *testing.T) {
 	}
 }
 
+func TestRequestModifier(t *testing.T) {
+	var got *rest.Request
+	b := NewBuilder(restmapper, LegacyCategoryExpander, scheme.Scheme, fakeClientWith("test", t, nil), corev1Codec).
+		NamespaceParam("foo").
+		ModifyRequests(func(req *rest.Request) {
+			got = req
```

Comment:
This is a good test, but it doesn't check that the client actually sees modifications to the requests.  e.g. if this was pass-by-value instead of reference, then this test wouldn't catch that.

Reply:
If what was pass by value?  `*rest.Request`?

Reply:
Rereading this I think I was wrong since we are comparing the pointers later on, so it this does guarantee that the request is the same value as passed to the modifier.